source/postprocess/global_statistics.cc

```diff
@@ -0,0 +1,296 @@
+    {
+      // set global statistics about this time step
+      statistics.add_value("Time step number", this->get_timestep_number());
+
+      if (this->get_parameters().convert_to_years == true)
+        {
+          statistics.add_value("Time (years)", this->get_time() / year_in_seconds);
+          statistics.set_precision("Time (years)", 12);
+          statistics.set_scientific("Time (years)", true);
+
+          statistics.add_value("Time step size (years)", this->get_timestep() / year_in_seconds);
+          statistics.set_precision("Time step size (years)", 12);
+          statistics.set_scientific("Time step size (years)", true);
+        }
+      else
+        {
+          statistics.add_value("Time (seconds)", this->get_time());
+          statistics.set_precision("Time (seconds)", 12);
+          statistics.set_scientific("Time (seconds)", true);
+
+          statistics.add_value("Time step size (seconds)", this->get_timestep());
+          statistics.set_precision("Time step size (seconds)", 12);
+          statistics.set_scientific("Time step size (seconds)", true);
+        }
+
+      // set global statistics about the mesh and problem size
+      statistics.add_value("Number of mesh cells",
+                           this->get_triangulation().n_global_active_cells());
+
+      unsigned int n_stokes_dofs = this->introspection().system_dofs_per_block[0];
+      if (this->introspection().block_indices.velocities != this->introspection().block_indices.pressure)
+        n_stokes_dofs += this->introspection().system_dofs_per_block[this->introspection().block_indices.pressure];
+
+      statistics.add_value("Number of Stokes degrees of freedom", n_stokes_dofs);
+      statistics.add_value("Number of temperature degrees of freedom",
+                           this->introspection().system_dofs_per_block[this->introspection().block_indices.temperature]);
+      if (this->get_parameters().n_compositional_fields > 0)
+        statistics.add_value("Number of degrees of freedom for all compositions",
+                             this->get_parameters().n_compositional_fields
+                             * this->introspection().system_dofs_per_block[this->introspection().block_indices.compositional_fields[0]]);
+    }
+
+
+    template <int dim>
+    void
+    GlobalStatistics<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Global statistics");
+        {
+          prm.declare_entry ("Write statistics for all nonlinear iterations", "false",
+                             Patterns::Bool (),
+                             "Whether to put every nonlinear iteration into a separate "
+                             "line in the statistics file (if true), or to output only "
+                             "one line per time step that contains the total number of "
+                             "linear iterations summed up over all nonlinear iterations.");
```

Comment:
-> iterations of the Stokes and advection linear system solvers summed up over ...